### ISSUE TITLE
Better toc generation for epub file

### ISSUE THREAD

**Issue by bennyyip** [BasioMeusPuga/Lector#80]
I got an epub file(some novels in Chinese, can be download from [here](https://drive.google.com/file/d/1hXhVPqZLb5ezMiY7Yz4_MyaD5f07q4x7/view?usp=sharing)) with nested toc structure which lector cannot parse it correctly. Lector only shows me a toc of numbers, but the actul files structure and toc looks like:
```bash
λ tree | rg -v "(html)|(jpe?g)|css|epub"
.
├── 1
│   ├── content.opf
│   ├── images
│   ├── text
│   └── toc.ncx
├── 2
│   ├── content.opf
│   ├── text
│   └── toc.ncx
....

├── 6
│   ├── content.opf
│   └── toc.ncx
├── content.opf
├── META-INF
│   ├── calibre_bookmarks.txt
│   └── container.xml
├── mimetype
├── toc.ncx
```
![image](https://user-images.githubusercontent.com/7467687/49696339-ba818b00-fbe3-11e8-8e9d-6d6ab48b6371.png)

I noticed that here(https://github.com/BasioMeusPuga/Lector/blob/master/lector/readers/read_epub.py#L213) gets a wrong toc file(`1/toc.ncx`, but should be `toc.ncx` in root).

There is an actively maintained epub parser libary(https://github.com/aerkalov/ebooklib) that could handle toc correctly. Maybe we can replace the simple parser in `readers/read_epub` with `ebooklib`.

**Comment by BasioMeusPuga** [comment]
I'll have a look at the files you mentioned. I'll probably have to redo the way `toc.ncx` files are looked at. 
As far as nested structures go, there's very little changing how they're going to be displayed within the program. I can probably put in some additional formatting within the drop down for a sub-chapter.

Speaking of that library, I actually used it in the initial days of writing this. I'm afraid it has way too many inconsistencies. The amount of code I had to write just to make the library work with a fraction of the epub structure variants exceeded the whole parser I have currently.
